internal/auth/credentials_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"fmt"
+	"testing"
+
+	googleAuth "cloud.google.com/go/auth"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"cloud.google.com/go/auth/credentials"
+)
+
+////////////////////////////////////////////////////////////////////////
+// Mocks
+////////////////////////////////////////////////////////////////////////
+
+type MockDetectCredentials struct {
+	mock.Mock
+}
+
+func (m *MockDetectCredentials) DetectDefault(opts *credentials.DetectOptions) (*googleAuth.Credentials, error) {
+	args := m.Called(opts)
+	if args.Get(0) == nil {
+		return nil, args.Error(1)
+	}
+	return args.Get(0).(*googleAuth.Credentials), args.Error(1)
+}
+
+// //////////////////////////////////////////////////////////////////////
+// Boilerplate
+// //////////////////////////////////////////////////////////////////////
+type credentialsTestSuite struct {
```

Comment:
Do we need suite? If not then please avoid, as it unnecessarily increases complexity by introducing an additional layer of abstraction.